patches/pluginlib.patch

```diff
@@ -12,31 +30,87 @@
  namespace
  {
  std::vector<std::string> catkinFindLib()
-@@ -103,14 +108,14 @@ ClassLoader<T>::ClassLoader(
+@@ -103,6 +108,8 @@ ClassLoader<T>::ClassLoader(
  {
    ROS_DEBUG_NAMED("pluginlib.ClassLoader", "Creating ClassLoader, base = %s, address = %p",
      base_class.c_str(), this);
++
 +#ifndef ANDROID
    if (ros::package::getPath(package_).empty()) {
      throw pluginlib::ClassLoaderException("Unable to find package: " + package_);
    }
-+#endif
- 
--  if (0 == plugin_xml_paths_.size()) {
--    plugin_xml_paths_ = getPluginXmlPaths(package_, attrib_name_);
--  }
--  classes_available_ = determineAvailableClasses(plugin_xml_paths_);
+@@ -111,6 +118,11 @@ ClassLoader<T>::ClassLoader(
+     plugin_xml_paths_ = getPluginXmlPaths(package_, attrib_name_);
+   }
+   classes_available_ = determineAvailableClasses(plugin_xml_paths_);
++#else
 +  // get static info of the already loaded plugin library
 +  classes_available_ = getStaticClassesAvailable();
++#endif
++
    ROS_DEBUG_NAMED("pluginlib.ClassLoader",
      "Finished constructring ClassLoader, base = %s, address = %p",
      base_class.c_str(), this);
-@@ -615,7 +620,7 @@ void ClassLoader<T>::loadLibraryForClass(const std::string & lookup_name)
+@@ -134,11 +146,13 @@ T * ClassLoader<T>::createClassInstance(const std::string & lookup_name, bool au
+     "In deprecated call createClassInstance(), lookup_name = %s, auto_load = %i.",
+     (lookup_name.c_str()), auto_load);
+ 
++#ifndef ANDROID
+   if (auto_load && !isClassLoaded(lookup_name)) {
+     ROS_DEBUG_NAMED("pluginlib.ClassLoader",
+       "Autoloading class library before attempting to create instance.");
+     loadLibraryForClass(lookup_name);
+   }
++#endif
+ 
+   try {
+     ROS_DEBUG_NAMED("pluginlib.ClassLoader",
+@@ -161,9 +175,11 @@ boost::shared_ptr<T> ClassLoader<T>::createInstance(const std::string & lookup_n
+   ROS_DEBUG_NAMED("pluginlib.ClassLoader", "Attempting to create managed instance for class %s.",
+     lookup_name.c_str());
+ 
++#ifndef ANDROID
+   if (!isClassLoaded(lookup_name)) {
+     loadLibraryForClass(lookup_name);
+   }
++#endif
+ 
+   try {
+     std::string class_type = getClassType(lookup_name);
+@@ -193,9 +209,11 @@ UniquePtr<T> ClassLoader<T>::createUniqueInstance(const std::string & lookup_nam
+     "Attempting to create managed (unique) instance for class %s.",
+     lookup_name.c_str());
+ 
++#ifndef ANDROID
+   if (!isClassLoaded(lookup_name)) {
+     loadLibraryForClass(lookup_name);
+   }
++#endif
+ 
+   try {
+     std::string class_type = getClassType(lookup_name);
+@@ -225,9 +243,11 @@ T * ClassLoader<T>::createUnmanagedInstance(const std::string & lookup_name)
+   ROS_DEBUG_NAMED("pluginlib.ClassLoader", "Attempting to create UNMANAGED instance for class %s.",
+     lookup_name.c_str());
+ 
++#ifndef ANDROID
+   if (!isClassLoaded(lookup_name)) {
+     loadLibraryForClass(lookup_name);
+   }
++#endif
```

Comment:
Hmm what if we patch `isClassLoaded` to return `true` always instead of adding so many `ifndef`?

Reply:
Maybe. Also, we could remove the old patch inside loadLibraryForClass().